### PR TITLE
Add sqlite date format

### PR DESCRIPTION
**Pull request by winkbrace**
This PR is a little strange. Since SQLite doesn't natively have a date_format function, I made a best effort to convert the Mysql `date_format` formats to the Sqlite `strftime` date formats. Unfortunately Sqlite only supports the [most common numeric formats](https://www.sqlite.org/lang_datefunc.html), so a lot might get lost in translation.

I still would like this function in the library however, because it will work and allow me to switch between sqlite and mysql without having to alter my code.

What do you think?

### CODE REVIEW

File: src/Query/Sqlite/DateFormat.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<?php namespace DoctrineExtensions\Query\Sqlite;
```
Comment (stevelacey) [inline src/Query/Sqlite/DateFormat.php]: `\n`

**Comment by stevelacey** [comment]
I am happy to consider this but a little hesitant as I don't fully understand the problem from the files changed. I'd be more comfortable if you had more tests of the scenarios you're catering for?

**Comment by winkbrace** [comment]
I expanded the test suite, thoroughly checked the conversions and fixed the mistakes, and put a little more information in the readme note.

Regarding your question, date_format() does not store datetimes, it converts datetimes to string representations in the format provided. 

This PR adds a function that does not natively exist in sqlite and translates it as good as possible.

**Comment by stevelacey** [comment]
Yes, it's very annoying, I've given it a kick

**Comment by stevelacey** [comment]
Thanks!

**Pull request merged**